sawmill-core/src/main/java/io/logz/sawmill/conditions/IpCompareCondition.java

```diff
@@ -0,0 +1,111 @@
+package io.logz.sawmill.conditions;
+
+import io.logz.sawmill.Condition;
+import io.logz.sawmill.Doc;
+import io.logz.sawmill.annotations.ConditionProvider;
+import io.logz.sawmill.parser.ConditionParser;
+import io.logz.sawmill.utilities.JsonUtils;
+
+
+import java.net.*;
+import java.util.*;
+
+import static java.util.Objects.requireNonNull;
+
+@ConditionProvider(type = "ipRange", factory = IpCompareCondition.Factory.class)
+public class IpCompareCondition implements Condition  {
+	
+	private String ipHigh;
+	private String ipLow;
+	private String ipTest;
+	
+	public IpCompareCondition(String ipTest, String ipLow, String ipHigh) {
+		this.ipTest = requireNonNull(ipTest);
+		this.ipHigh = requireNonNull(ipHigh);
+		this.ipLow = requireNonNull(ipLow);
+		
+	}
+	
+	public static long ipToLong(InetAddress ip) {
+        byte[] octets = ip.getAddress();
+        long result = 0;
+        for (byte octet : octets) {
+            result <<= 8;
+            result |= octet & 0xff;
+        }
+        return result;
+    }
+	
+    @Override
+    public boolean evaluate(Doc doc) {
```

Comment:
@DanMelman Okay I think I understand but I am a bit confused on how to get it to work.
So I need to add ipTest, ipHigh and ipLow to the doc so I can use something like
(String)doc.getField(ipTest); in the evaluate function?
However I am not sure how to add them to doc.
Like I am looking at
https://github.com/logzio/sawmill/blob/master/sawmill-core/src/main/java/io/logz/sawmill/conditions/InCondition.java#L31
and I can see they are grabbing the path field from the doc
Object pathValue = doc.getField(path);
But I am not seeing anything that is adding the path field to the doc. What am I missing?

Reply:
let's explain the `InCondition` code, it might make things more clear to you:
1. The condition is created by it's `Factory`, based on the definition of the pipeline. That definition contains two parameters - `path` and `value`.
2. While processing a document, we try to get the field in the defined `path` by calling `doc.getField(path)` (after we validate it exists, using `doc.hasField(path)`)
3. The actual value of the field in that `path` is stored in a variable called `pathValue`, which is a list that is then used to check whether the defined `value` exists in that list. If it does, `anyMatch` will return `true`, otherwise `false`.

Reply:
@DanMelman 
Okay ill read over this and see if I can figure it out thanks.

Though I did write the test in my eclipse workspace and it is working there. lol I don't want you to think I just added the code here without testing it first

```
public class Test {

	public static void main(String[] args) {
		// TODO Auto-generated method stub
        Pipeline pipeline = new Pipeline.Factory().create(
        				"{\n" +
        				"  \"steps\": [\n" +
        				"    {\n" +
        				"      \"if\": {\n" +
        				"        \"condition\": {\n" +
        				"          \"ipRange\": {\n" +
        				"            \"ipHigh\": \"192.255.0.0\"\n" +
        				"            \"ipTest\": \"192.200.3.0\",\n" +
        				"            \"ipLow\": \"192.0.0.0\"\n" +
        				"          }\n" +
        				"        },\n" +
        				"        \"then\": [\n" +
        				"          {\n" +
        				"            \"addField\": {\n" +
        				"              \"config\": {\n" +
        				"                \"path\": \"testing\",\n" +
        				"                \"value\": \"itWorked\"\n" +
        				"              }\n" +
        				"            }\n" +
        				"          }\n" +
        				"        ]\n" +
        				"      }\n" +
        				"    }\n" +
        				"  ]\n" +
        				"}");

        Doc doc = createDoc("message", "testing geoip resolving", "ip", "192.200.3.0", "test", "a");
        ExecutionResult executionResult = new PipelineExecutor().execute(pipeline, doc);

        if (executionResult.isSucceeded()) {
            System.out.println("Success! result is:"+doc.toString());
        }

	}

}
```
3234333440 192.200.3.0 test
3221225472 192.0.0.0 low
3237937152 192.255.0.0 high
true
Success! result is:Doc{source={message=testing geoip resolving, ip=192.200.3.0, test=a, testing=itWorked}}

Reply:
@DanMelman 
It doesn't in my test which is confusing to me. For example if I change 
ipTest to 194.200.3.0 (which is higher than ipHigh). It returns false and the testing field 'itWorked' is not added to the doc. So to me it seemed like everything is working. But obviously it is not, as String)doc.getField(ipHigh) returns errors in the evaluate function. Thanks for all your help so far

3267887872 194.200.3.0 test
3221225472 192.0.0.0 low
3237937152 192.255.0.0 high
false
Success! result is:Doc{source={message=testing geoip resolving, ip=192.200.3.0, test=a}}

Reply:
I really suggest you try to create a test like InConditionTest for your code, with all flows (returning true, false, invalid field, etc..)

Reply:
Then you could debug the failing tests and fix the code

Reply:
Thank will do

Reply:
Okay changed the 3 lines below. See comments for what changed